File: algo/sorting/lc_97_K_Closest_Points_to_Origin/python/Solution1.py
class Solution1:

    def kCloest(self, points: List[List[int]], k: int) -> List[List[int]]:
        points.sort(key = self.square_distance)
        return points[:k]

    def square_distance(self, point: List[int]) -> int:
        return point[0]**2 + point[1]**2

File: algo/sorting/lc_97_K_Closest_Points_to_Origin/java/Solution1.java
class Solution1 {
    // Time Complexity: O(NLogN)
    public int[][] kClosest(int[][] points, int k) {
        Arrays.sort(points, (a, b) -> squareDistance(a) - squareDistance(b));
        return Arrays.copyOf(points, 0, k);
    }

    int squareDistance(int[] point) {
        return point[0] * point[0] + point[1] * point[1];
    }
}